geemdjem0819/Assets/scripts/PlantBranch.cs

```diff
@@ -0,0 +1,184 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public struct Ring
+{
+    private List<Vector3> vertices;
+
+    public Vector3 position;
+    public Quaternion rotation;
+    public Vector3 scale;
+
+    public Ring(Transform transform, float radius, int nVertices)
+    {
+        vertices = new List<Vector3>();
+        position = transform.position;
+        rotation = transform.rotation;
+        scale = transform.localScale;
+
+        for (int i = 0; i < nVertices; i++) {
+            var t = 2 * Mathf.PI * i / nVertices;
+
+            var x = radius * Mathf.Cos(t);
+            var z = radius * Mathf.Sin(t);
+
+            var vertex = new Vector3(x, 0, z);
+
+            vertices.Add(vertex);
+        }
+    }
+
+    /// Retrieve transformed vertices from this Ring.
+    public List<Vector3> Vertices
+    {
+        get {
+            var matrix = Matrix4x4.TRS(position, rotation, scale);
+
+            var transformedVertices = new List<Vector3>();
+
+            foreach (Vector3 vector in vertices) {
+                transformedVertices.Add(matrix.MultiplyPoint3x4(vector));
+            }
+
+            return transformedVertices;
+        }
+        private set { vertices = value; }
+    }
+}
+
+[RequireComponent(typeof(MeshFilter))]
+public class PlantBranch : MonoBehaviour
+{
+    public int nVertices = 8;
+
+    public float branchRadius = 1f;
+    public float startRadius = 1f;
+    public float tipFalloff = 5f;
+
+    public float segmentLength = 1f;
+    public float growSpeed = 1f;
+    public float turnSpeed = 3f;
+
+    public float directionVariance = 1f;
+
+    public Transform growTarget;
+
+    // Global direction of growth.
+    private Transform growBase;
+    // Last position (local) where a branch segment has been spawned.
+    private Vector3 lastPosition;
+    private Quaternion growDirection;
+
+    private bool isGrowing = true;
+    public bool IsGrowing { get; set; }
```

Comment:
kssssst, wegwezen met deze... of die andere... of niet, het is een vrij land, zoek het uit. Abe.